commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -474,36 +475,47 @@ def update_return_payment_state_after_successful_refund(
-                return_item_id=return_line_item_return_id,
-                payment_state=ReturnPaymentState.REFUNDED,
+                f"with ids {return_line_item_return_ids} to '{ReturnPaymentState.REFUNDED}'."
             )
             if not payment_intent_id:
                 payment_intent_id = ""
             logger.info(f"Creating return for order - payment_intent_id: {payment_intent_id}")
             payment = self.get_payment_by_key(payment_intent_id)
             logger.info(f"Payment found: {payment}")
             transaction_id = find_refund_transaction(payment, amount_in_cents)
-            update_transaction_id_action = OrderSetReturnItemCustomTypeAction(
-                return_item_id=return_line_item_return_id,
-                type=CTTypeResourceIdentifier(
-                    key="returnItemCustomType",
-                ),
-                fields=CTFieldContainer({"transactionId": transaction_id}),
-            )
+            return_payment_state_actions = []
+            update_transaction_id_actions = []
+            for return_line_item_return_id in return_line_item_return_ids:
+                return_payment_state_actions.append(OrderSetReturnPaymentStateAction(
+                    return_item_id=return_line_item_return_id,
+                    payment_state=ReturnPaymentState.REFUNDED,
+                ))
+                custom_fields = {
+                    "transactionId": transaction_id,
+                }
+                entitlement_id = return_line_entitlement_ids.get(return_line_item_return_id)
+                if entitlement_id:
+                    custom_fields[TwoUKeys.LINE_ITEM_LMS_ENTITLEMENT_ID] = entitlement_id
+                update_transaction_id_actions.append(OrderSetReturnItemCustomTypeAction(
+                    return_item_id=return_line_item_return_id,
+                    type=CTTypeResourceIdentifier(
+                        key="returnItemCustomType",
+                    ),
+                    fields=CTFieldContainer(custom_fields),
+                ))
+
             return_transaction_return_item_action = PaymentSetTransactionCustomTypeAction(
                 transaction_id=transaction_id,
                 type=CTTypeResourceIdentifier(key="transactionCustomType"),
-                fields=CTFieldContainer({"returnItemId": return_line_item_return_id}),
+                # TODO: ask shafqat
+                fields=CTFieldContainer({"returnItemId": return_line_item_return_ids[0]}),
             )
             logger.info(f"Update return payment state after successful refund - payment_intent_id: {payment_intent_id}")
 
             updated_order = self.base_client.orders.update_by_id(
                 id=order_id,
                 version=order_version,
-                actions=[return_payment_state_action, update_transaction_id_action],
+                actions=return_payment_state_actions + update_transaction_id_actions,
```

Comment:
why did we change this?

Reply:
previously there was one item to return, for which one payment state update and one transaction id update action was called. Now we have multiple lines so we'll get a list and i am just combining both these lists here.